src/container/__tests__/analytics.test.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+
+import Container from '../../../lib/components/container';
+
+import { FunnelMetrics, setFunnelMetrics } from '../../../lib/components/internal/analytics';
+
+import {
+  AnalyticsFunnel,
+  AnalyticsFunnelStep,
+} from '../../../lib/components/internal/analytics/components/analytics-funnel';
+
+const mockedFunnelInteractionId = 'mocked-funnel-id';
+function mockFunnelMetrics() {
+  setFunnelMetrics({
+    funnelStart: jest.fn(() => mockedFunnelInteractionId),
+    funnelError: jest.fn(),
+    funnelComplete: jest.fn(),
+    funnelSuccessful: jest.fn(),
+    funnelCancelled: jest.fn(),
+    funnelStepStart: jest.fn(),
+    funnelStepComplete: jest.fn(),
+    funnelStepNavigation: jest.fn(),
+    funnelSubStepStart: jest.fn(),
+    funnelSubStepComplete: jest.fn(),
+    funnelSubStepError: jest.fn(),
+    helpPanelInteracted: jest.fn(),
+    externalLinkInteracted: jest.fn(),
+  });
+}
+
+describe('Funnel Analytics', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockFunnelMetrics();
+  });
+
+  test('renders normally when outside the context of a funnel', () => {
+    render(<Container />);
+    // No error occurred if the component rendered successfully
```

Comment:
This is not really a test case without any assertions. Is there something we can `expect` here?

Reply:
Added an assertion. This also came with some brought some new changes to not add any of the attributes if outside of the context.